examples/angularjs/js/services/todoStorage.js

```diff
@@ -151,8 +151,12 @@ angular.module('todomvc')
 
 			put: function (todo, index) {
 				var deferred = $q.defer();
-
-				store.todos[index] = todo;
+				// So that a item that was removed before the image was fetched
+				// is not added again.
+				const found = store.todos.indexOf(todo);
```

Comment:
This change broke the function. If we cal `put` with a modified `todo` item instance, it will never ever be found with `indexOf` just because it has been modified. The `index` parameter was here on purpose. 

Having `index` parameter was a pretty bad design decision from the authors of this example. The other `storage` implementation in this file does not require this parameter so there is a discrepancy in interface between the two implementations.